src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -99,19 +100,6 @@ def get_current_value(field_name):
 
         return get_current_value
 
-    def validate_name(self, value):
```

Comment:
@nolachen If possible, I think we should put the check in here instead of in `validate` which is getting huge. I think there might be some challenges with that because you need to access `isInternal` but since that field can never change in value, I think we can find a way around it.

Reply:
@scefali do you know of any way to do that? it seems like you can't access other fields from a field-level validator in DRF

https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation
https://stackoverflow.com/questions/36549574/get-value-of-another-field-in-field-level-validation-in-drf

Reply:
@nolachen I remember this. What about looking in `kwargs["data"]["isInternal"]` or in `self.instance`?

Reply:
I honestly hate how Django doesn't let you access `attrs` outside of `validate` so I am OK with a workaround that let's us do it even if it's just like `self.attrs = `kwargs["data"]`. I would refactor some of other mess myself if we can get that done.